docs/howto/csi.md

```diff
@@ -0,0 +1,101 @@
+# Registering a CSI Driver
+
+For more information on what is the Container Storage Interface and how it
+relates to a CSI driver, see the [topic guide](../topics/csi.md) for more
+information.
+
+A Krustlet Provider with CSI support will check for new drivers registered to
+the `plugins/` directory (by default, this is `$HOME/.krustlet/plugins`). You
+will need to inform your CSI driver to bind its socket at that location in order
+for a Provider to recognize and register the driver.
+
+At this time, you will also need to install and run the following projects so
+that the PersistentVolumeClaim's volume will be provisioned and readily
+available for use:
+
+- [node-driver-registrar](https://github.com/kubernetes-csi/node-driver-registrar)
+- [external-provisioner](https://github.com/kubernetes-csi/external-provisioner)
+
+Do keep in mind that these projects rely on linux-specific command line tooling
+like `mount`, so these tools may only work on Linux. MacOS and Windows support
+is not guaranteed.
+
+These projects will eventually be merged into the Krustlet project in future
```

Comment:
Only external provisioner will be I think. The registrar is something that should be built into CSI implementations or just leverage the existing project

Reply:
So after investigating more, I don't think we have any immediate plans to port these over. Looks like external-provisioner is not linux specific. The node-driver-registrar is OS specific and we probably need to contribute back to it so it will work on Mac (which is a really easy fix)

Reply:
Opened a PR to allow for building on Darwin: https://github.com/kubernetes-csi/node-driver-registrar/pull/133